src/core/public/chrome/doc_title/doc_title_service.ts

```diff
@@ -0,0 +1,180 @@
+ * ```
+ *
+ * @example
+ * How to reset the title of the document to it's initial value
+ * ```ts
+ * chrome.docTitle.reset()
+ * ```
+ *
+ * @public
+ * */
+export interface ChromeDocTitle {
+  /**
+   * Gets an observable of the current document title.
+   */
+  get$: () => Observable<string>;
+  /**
+   * Changes the current document title.
+   *
+   * @remarks
+   * The apply option is mostly here for legacy compatibility reasons and it's default value
+   * should probably always be used.
+   *
+   * @example
+   * How to change the title of the document
+   * ```ts
+   * chrome.docTitle.change('My application title')
+   * chrome.docTitle.change(['My application', 'My section'])
+   * chrome.docTitle.change({ parts: ['My application'], excludeBase: true })
+   * ```
+   *
+   * @param newTitle The new title to set, either a string, string array or {@link ChromeDocTitleEntry}
+   * @param [apply=true] If false, will not actually apply the new title until #apply() is called.
+   */
+  change: (newTitle: ChromeDocTitleChange, apply?: boolean) => void;
+  /**
+   * Resets the document title to it's initial value.
+   * (meaning the one present in the title meta at application load.)
+   *
+   * @remarks
+   * The apply option is mostly here for legacy compatibility reasons and it's default value
+   * should probably always be used.
+   *
+   * @param [apply=true] If false, will not actually apply the new title until #apply() is called.
+   */
+  reset: (apply?: boolean) => void;
+  /**
+   * Apply a title change or reset that was not applied yet.
+   *
+   * @see {DocTitle#change}
+   * @see {DocTitle#reset}
+   */
+  apply: () => void;
+  /** @internal */
+  __legacy: {
+    setBaseTitle: (baseTitle: string) => void;
+  };
+}
+
+/** @public */
+export interface ChromeDocTitleEntry {
+  parts: string[];
+  excludeBase?: boolean;
+}
+
+/**
+ * Composed type for the {@link ChromeDocTitle#change} possible inputs.
+ *
+ * @public
+ */
+export type ChromeDocTitleChange = string | string[] | ChromeDocTitleEntry;
+
+const defaultTitle: ChromeDocTitleEntry = { parts: [], excludeBase: false };
+
+const inputToEntry = (input: ChromeDocTitleChange): ChromeDocTitleEntry => {
+  if (isString(input)) {
+    return {
+      parts: [input],
+      excludeBase: false,
+    };
+  }
+  if (isArray<string>(input)) {
+    return {
+      parts: [...input],
+      excludeBase: false,
+    };
+  }
+  return input;
+};
+
+/** @internal */
+export class DocTitleService {
+  private document = { title: '' };
+  private baseTitle = '';
+  private current = defaultTitle;
+  private readonly title$ = new BehaviorSubject<string>('');
+  private readonly stop$ = new ReplaySubject(1);
+
+  public start({ document }: StartDeps): ChromeDocTitle {
+    this.document = document;
+    this.baseTitle = document.title;
+    this.title$.next(this.baseTitle);
+
+    return {
+      get$: () => this.title$.pipe(takeUntil(this.stop$)),
```

Comment:
What actually uses this?

Reply:
No one atm. Just thought it might be useful to expose it so its already present if a plugin may want to use it in the futur. Do you think it's better to just remove it ?